### PR TITLE
Adding quicklogic tests and updating the corresponding conf file to r…

### PR DESCRIPTION
**Pull request by lnsharma**
…un them

---
name: Pull request
about: Push a change to this project
---

### Motivate of the pull request
- [ ] To address an existing issue. If so, please provide a link to the issue.
- [ ] Breaking new feature. If so, please decribe details in the description part.
@lnsharma : Adding new test cases to test quicklogic flow. Currently, all these tests run is commented as openfpga_shell script is failing in iverilog for the already present counter design. This needs to be fixed first before we can uncomment all these tests.

### Describe the technical details
- What is currently done? (Provide issue link if applicable)
- What does this pull request change?

### Which part of the code base require a change
**In general, modification on existing submodules are not acceptable. You should push changes to upstream.**
- [ ] VPR
- [ ] OpenFPGA libraries
- [ ] FPGA-Verilog
- [ ] FPGA-Bitstream
- [ ] FPGA-SDC
- [ ] FPGA-SPICE
- [ ] Flow scripts
- [ ] Architecture library
- [ ] Cell library

### Checklist of the pull request
- [ ] Require code changes. 
- [ ] Require new tests to be added
- [ ] Require an update on documentation

### Impact of the pull request
- [ ] Require a change on Quality of Results (QoR)
- [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

**Comment by ganeshgore** [comment]
@lnsharma  Master branch was failing the regression test, It has been patched just now.
Please wait for regression test on master to finish, and rerun this regression from actions tab.

**Comment by ganeshgore** [comment]
@lnsharma You will have to merge current master

**Comment by tangxifan** [comment]
@lnsharma Here are the notes from our discussion
- Remove the micro benchmarks ``and2`` etc. because they are already in the OpenFPGA repo
- Disable the end-flow test in task configuration file
- I will build a test case for the 5-clock benchmark
- I will build a test case for the adder-LUT benchmark as well as architecture

**Comment by lnsharma** [comment]
> @lnsharma Here are the notes from our discussion
> 
> * Remove the micro benchmarks `and2` etc. because they are already in the OpenFPGA repo
> * Disable the end-flow test in task configuration file
> * I will build a test case for the 5-clock benchmark
> * I will build a test case for the adder-LUT benchmark as well as architecture

@lnsharma : In my latest checkin, I uncommenting all tests except 2 that requires multiple clocks. I think now you can approve this request once CI passes.

File: openfpga_flow/benchmarks/quicklogic_tests/and2/and2.act
Comment (tangxifan) [inline openfpga_flow/benchmarks/quicklogic_tests/and2/and2.act:2]: @lnsharma Can you remove these duplicated micro benchmarks?

File: openfpga_flow/benchmarks/quicklogic_tests/i2c_master_top/i2c_master_top_yosys.blif
Comment (tangxifan) [inline openfpga_flow/benchmarks/quicklogic_tests/i2c_master_top/i2c_master_top_yosys.blif:1]: @lnsharma  I see some benchmarks have .blif files pushed. Are they required or they can be reproduced by yosys?

**Review by tangxifan — CHANGES_REQUESTED** [review]
@lnsharma The PR is close to merge. Just need some clean up.

File: openfpga_flow/benchmarks/quicklogic_tests/i2c_master_top/i2c_master_top_yosys.blif
Comment (lnsharma) [reply openfpga_flow/benchmarks/quicklogic_tests/i2c_master_top/i2c_master_top_yosys.blif:1]: I removed the blif files as well as and2 test case.

File: openfpga_flow/benchmarks/quicklogic_tests/and2/and2.act
Comment (lnsharma) [reply openfpga_flow/benchmarks/quicklogic_tests/and2/and2.act:2]: Removed

**Review by tangxifan — COMMENTED** [review]
@lnsharma Can you double-check if you have pushed the commits to github? From my side, I can still see the micro benchmark ``and2``, ``and2_latch``, ``and2_or2``, routing_test`` as well as ``.blif`` files

**Comment by tangxifan** [comment]
This is what I see now.

image

**Comment by lnsharma** [comment]
> @lnsharma Can you double-check if you have pushed the commits to github? From my side, I can still see the micro benchmark `and2`, `and2_latch`, `and2_or2`, routing_test`as well as`.blif`` files

please check now

**Review by tangxifan — APPROVED** [review]

**Comment by tangxifan** [comment]
@lnsharma Looks good now. Just need to see the CI passed.

**Comment by tangxifan** [comment]
It seems that the flow-run requires more 14GB disk space, beyond a Github action runner can afford. 
We have see similar problems in SOFA repo.
The solution is not to output Verilog testbenches unless they are needed.
You can consider to use the [openfpga shell script](https://github.com/lnis-uofu/OpenFPGA/blob/master/openfpga_flow/openfpga_shell_scripts/generate_bitstream_example_script.openfpga) which is tuned to generate bitstreams only

**Comment by tangxifan** [comment]
@lnsharma Please be aware that the regression tests failed currently. We have to fix that before merging.

**Comment by lnsharma** [comment]
> @lnsharma Please be aware that the regression tests failed currently. We have to fix that before merging.



> @lnsharma Please be aware that the regression tests failed currently. We have to fix that before merging.

Just checked that in OpenFPGA, quicklogic tests that I added, all are passing successfully. Do you mean that some other tests are failing?

**Pull request merged**
